### PR TITLE
Fix provenance logic for jax 0.2.28

### LINKED ISSUE #1318: `numpyro.render_model` behaviour changed in 0.9.0
I noticed that the behaviour of `numpyro.render_model` appears to have changed since the latest release, specifically arrows signifying conditional dependence are not being rendered properly.

I haven't had time to dig into why this is yet, but here is a simple recreation

```python
import numpyro
import numpyro.distributions as dist


def model():
    theta = numpyro.sample("theta", dist.Uniform())
    
    with numpyro.plate("N", size=10):
        numpyro.sample("y", dist.Bernoulli(theta))
        
        
if __name__ == "__main__":
    g = numpyro.render_model(model)
    g.render(f"model-{numpyro.__version__}", format="png")
```

Running this with 0.8.0 and 0.9.0 I get (respectively)

model-0 8 0

model-0 9 0

### PR DESCRIPTION
**Pull request by fehiepsi**
Fixes #1318

tldr; I have tested the change and it works for the old jax versions.

The recent jax release made provenance logic failing because the cached abstract value might be a different object w.r.t. the one stored in the output of `process_primitive`, while we only updated `provenance` for the output previously.

I think we'll need to make a patch release after this fix.

### CODE REVIEW

**Review by fritzo — APPROVED** [review]
Nice, that's subtle!

Do you think it's worth adding a regression test, e.g. the model added in the issue?

**Comment by fehiepsi** [comment]
Hi Fritz, all the tests involving provenance failed so we have to pin the jax version to make CI pass. In this pr, I just removed the restriction because the tests pass again.

**Comment by fehiepsi** [comment]
Thanks for reviewing! :)
